packages/toolpad-app/src/server/auth.ts

```diff
@@ -27,6 +31,42 @@ export function createAuthHandler(base: string): Router {
           GithubProvider({
```

Comment:
Does this provider have to be recreated on every request? Would it make sense to put it in its own module?

Reply:
I was just following some examples and didn't consider that, it should be better to move it outside, will do!

Reply:
Things didn't work when I moved the provider to a separate variable, I think it should be fine to do it like this as that's how the examples I have seen work too.

Reply:
> Things didn't work when I moved the provider to a separate variable

Any idea why? This seems like a strange thing. Can you investigate?

Reply:
Sure I'm taking a look at how it works internally, this library has some weird things sometimes though, it seems to be a bit of a work in progress and most things aren't documented at all.

Reply:
Nevermind it wasn't about the library this time - I was trying to set the providers as global variables but I don't think the environment variables had been initialized there. If I set the variables inside the `createHandler` function it works this fine, this should be enough? https://github.com/mui/mui-toolpad/pull/3047/commits/39c0f006c403dd400f6d9611a81cd8766b29a422

Reply:
That's what I assumed. What's preventing us from initializing variables earlier? It feels very counterintuitive and error-prone to have half the program work with uninitialized environment

Reply:
Isn't the environment set with `this.envManager.start()`? This runs after the `auth.ts` file is imported, that's why it's not there for a global variable in that file.

Reply:
Maybe we can load the Toolpad app's environment before any imports when the server starts if you think it's worth it. Sounds like an unrelated issue though, I don't see what's wrong with initializing the providers in `createAuthHandler`.

Reply:
I can also load the providers in a global function if that's better.

Reply:
> Maybe we can load the Toolpad app's environment before any imports when the server starts if you think it's worth it

Just

```tsx
import 'dotenv/config'
```
before anything else

Reply:
Ok, it works if I add that line to `toolpadAppServer.ts`, as that file seems to run before `index.ts` - it might be because the worker imports that file first, I can look deeper into it and try in another file if this one doesn't seem ideal for any reason.
https://github.com/mui/mui-toolpad/pull/3047/commits/07015a045461e47395b092807171c3aec9b42277